hedera-node/src/main/java/com/hedera/services/store/contracts/precompile/codec/EncodingFacade.java

```diff
@@ -17,431 +12,660 @@
+            this.name = name;
+            return this;
+        }
+
+        private FunctionResultBuilder withSymbol(final String symbol) {
+            this.symbol = symbol;
+            return this;
+        }
+
+        private FunctionResultBuilder withTokenUri(final String tokenUri) {
+            this.metadata = tokenUri;
+            return this;
+        }
+
+        private FunctionResultBuilder withErcFungibleTransferStatus(
+                final boolean ercFungibleTransferStatus) {
+            this.ercFungibleTransferStatus = ercFungibleTransferStatus;
+            return this;
+        }
+
+        private FunctionResultBuilder withIsApprovedForAllStatus(
+                final boolean isApprovedForAllStatus) {
+            this.isApprovedForAllStatus = isApprovedForAllStatus;
+            return this;
+        }
+
+        private FunctionResultBuilder withTokenInfo(final TokenInfo tokenInfo) {
+            this.tokenInfo = tokenInfo;
+            return this;
+        }
+
+        private FunctionResultBuilder withNonFungibleTokenInfo(
+                final NonFungibleTokenInfo nonFungibleTokenInfo) {
+            this.nonFungibleTokenInfo = nonFungibleTokenInfo;
+            return this;
+        }
+
+        private Bytes build() {
+            final var result =
+                    switch (functionType) {
+                        case CREATE -> Tuple.of(
+                                status, convertBesuAddressToHeadlongAddress(newTokenAddress));
+                        case MINT -> Tuple.of(
+                                status, BigInteger.valueOf(totalSupply), serialNumbers);
+                        case BURN -> Tuple.of(status, BigInteger.valueOf(totalSupply));
+                        case TOTAL_SUPPLY -> Tuple.of(BigInteger.valueOf(totalSupply));
+                        case DECIMALS -> Tuple.of(decimals);
+                        case BALANCE -> Tuple.of(BigInteger.valueOf(balance));
+                        case OWNER -> Tuple.of(convertBesuAddressToHeadlongAddress(owner));
+                        case NAME -> Tuple.of(name);
+                        case SYMBOL -> Tuple.of(symbol);
+                        case TOKEN_URI -> Tuple.of(metadata);
+                        case ERC_TRANSFER -> Tuple.of(ercFungibleTransferStatus);
+                        case ALLOWANCE -> Tuple.of(BigInteger.valueOf(allowance));
+                        case APPROVE -> Tuple.of(approve);
+                        case GET_APPROVED -> Tuple.of(
+                                convertBesuAddressToHeadlongAddress(approved));
+                        case IS_APPROVED_FOR_ALL -> Tuple.of(isApprovedForAllStatus);
+                        case GET_TOKEN_INFO -> getTupleForGetTokenInfo();
+                        case GET_FUNGIBLE_TOKEN_INFO -> getTupleForGetFungibleTokenInfo();
+                        case GET_NON_FUNGIBLE_TOKEN_INFO -> Tuple.of(nonFungibleTokenInfo);
```

Comment:
The approach is inconsistent between fungible and non-fungible.

In the fungible case you pull out decimals earlier and use `getTupleForGetFungibleTokenInfo()` to build a tuple of tokenInfo and decimals.
In the non-fungible case your don't pull anything out earlier and create a single tuple here.

I think either approach is acceptable but the inconsistency makes it more challenging to maintain.
 
Maybe add a  `getTupleForGetNonFungibleTokenInfo()` since you're currently missing the status in the tuple anyway.
Should be the following at some stage right?
```
Tuple.of(status, Tuple.of(nonFungibleTokenInfo));
```

Reply:
You're correct. I've left the PR in an inconsistent state and haven't committed refactoring logic for the non-fungible case. It's updated now.